lib/parser/lexer.rl

```diff
@@ -1769,16 +1769,21 @@ class Parser::Lexer
       };
 
       # a([1, 2])
-      e_lbrack    |
+      e_lbrack
+      => { emit(:tLBRACK)
+           fbreak; };
+
       # a()
       e_lparen
-      => { emit_table(PUNCTUATION_BEGIN)
+      => { emit(:tLPAREN)
            fbreak; };
 
-      # a(+b)
-      punctuation_begin
-      => { emit_table(PUNCTUATION_BEGIN)
```

Comment:
A reader (like me) can see right away which constructs will be caught by this rule, and which tokens will be emitted. You don't have to jump back and forth between looking at the code here and looking at the punctuation tables. Also, `PUNCTUATION_BEGIN` is a big table, but only a few entries are relevant here.

But if you don't agree, that is fine.